clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -843,6 +843,61 @@ Block *BrCondOp::getSuccessorForOperands(ArrayRef<Attribute> operands) {
   return nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// CIR_BitOp family of operations
+//===----------------------------------------------------------------------===//
+
+LogicalResult BitClrsbOp::verify() {
+  if (!getInput().getType().isSigned())
+    return emitOpError("input of cir.bit.clrsb must be a signed integer");
+  if (!getType().isSigned() || getType().getWidth() != 32)
```

Comment:
- All these functions are doing the exact same check.
- The checks themselves can be done by adding constraints to tablegen (see `CIRTypes.td` and look for `UInt32`, `VoidPtr`, etc).

Reply:
Updated. Moved all verification code into TableGen.

Reply:
Awesome!